pkg/db/file_impl.go

```diff
@@ -56,4 +59,118 @@ func (F *FSBackend) Files() ([]string, error) {
 	return names, nil
 }
 
+//used for storing things in a file (remove if DB starts being used)
+//create a category-json file in the OS's temp directory
+var ex, _ = os.Executable()
+var exPath = filepath.Dir(ex)
+var categoriesFile = exPath + "/cat.json"
+
+func jsonFromFile(filename string, i interface{}) (err error) {
+	jsonFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(jsonFile, &i)
+
+	return err
+}
+func jsonToFile(filename string, i interface{}) error {
+	newJson, err := json.Marshal(i)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filename, newJson, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getCategoriesFromFile() (Categories, error) {
+	var cat Categories
+	err := jsonFromFile(categoriesFile, &cat)
+	return cat, err
+}
+
+func (F *FSBackend) CreateCategory(name string) (Category, error) {
+	// todo: use a db when worried about things like "concurrent requests"
+	//  category names should be unique (across documents?) either way,
+	//  to be enforced by the db via constraints (check those errors)
```

Comment:
Just to make sure, lets put a mutex in side the struct (m sync.Mutex), and lock it / unlock it on these operations.

Reply:
The critical section should probably just be any time you load or save the file.